tests/Session/test_session.py

```diff
@@ -1,14 +1,92 @@
-from src.Session.session import Session
+from src.Constants.commands_messages import INVALID_RESPONSE_MESSAGE  
+from src.Constants.constants import \
+    ADD_FOOD, \
+    CREATE_FOOD_STORAGE, \
+    CREATE_FREEZER, \
+    CREATE_FRIDGE, \
+    CREATE_PROFILE, \
+    DELETE_FOOD_STORAGE, \
+    DELETE_FREEZER, \
+    DELETE_FRIDGE, \
+    DELETE_PROFILE, \
+    EXIT, \
+    FOOD_STORAGE, \
+    FREEZER, \
+    FRIDGE, \
+    HELP, \
+    LIST_FOOD_STORAGES, \
+    LIST_FREEZERS, \
+    LIST_FRIDGES, \
+    LOAD, \
+    JSON_EXTENSION, \
+    NEW_LINE, \
+    OPEN_FOOD_STORAGE, \
+    OPEN_FREEZER, \
+    OPEN_FRIDGE, \
+    REMOVE_FOOD, \
+    SAVE
 from src.Constants.session_messages import *
+from src.Session.session import Session
 from unittest.mock import patch
 
+import sys, io
 import unittest
 
 class Test_Session(unittest.TestCase):
     def setUp(self):
         self.session = Session()
 
+    # Checks that the method located at method_path is called once with potential parameters
+    # expected_args when the user provides input within side_effect.
+    def checkCommandMethodCalled(self, mocked_input, side_effect, method_path, *expected_args):
+        mocked_input.side_effect = side_effect
+        with patch(method_path) as mock:
+            self.session.start()
+        mock.assert_called_once_with(*expected_args)
+
     @patch('src.Session.session.input', create=True)
     def testStart(self, mocked_input):
-        # TODO: This can't be tested right now until we do regression tests
-        pass
\ No newline at end of file
+        # This skips checking the welcome message, since twoPrintOutcomes
+        # has been tested in utils and it depends on someone's personal profiles.
+
+        # Testing that the 'exit' command properly exits the program
+        mocked_input.side_effect = [EXIT]
+        self.session.start()
+        self.assertRaises(SystemExit)
```

Comment:
Do you think you can explain how `assertRaises()` works? I tried looking it up in the documentation and it said it's to "Test that an exception is raised when callable is called with any positional or keyword arguments that are also passed to [assertRaises()] (https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)." which I don't really understand. Is exiting an 'exception' in this case?

Reply:
Yeah! I haven't read the documentation, but essentially it's checking that a specific exception is raised. So here, I'm checking that the SystemExit Exception is raised. 

In our case, this happens when the program ends from:
1. the user presses keys: `ctrl` + `c`
2. they type 'exit', which makes us break out of the while loop in `start()`, stopping naturally

Reply:
I see, thank you!!